x-pack/plugins/synthetics/public/apps/synthetics/components/monitor_add_edit/form/submit.tsx

```diff
@@ -33,7 +33,7 @@ export const ActionBar = () => {
     formState: { errors },
   } = useFormContext();
 
-  const [isDeleteModalVisible, setIsDeleteModalVisible] = useState(false);
+  const [isDeleteModalVisible, setIsDeleteModalVisible] = useState<SyntheticsMonitor | null>(null);
```

Comment:
I see so you are passing the config to avoid less prop passing down the line
```suggestion
  const [monitorPendingDeletion, setMonitorPendingDeletion] = useState<SyntheticsMonitor | null>(null);
```